apprise/plugins/NotifyPagerDuty.py

```diff
@@ -0,0 +1,438 @@
+# furnished to do so, subject to the following conditions :
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# API Refererence:
+#   - https://developer.pagerduty.com/api-reference/\
+#       368ae3d938c9e-send-an-event-to-pager-duty
+#
+
+import requests
+from json import dumps
+
+from .NotifyBase import NotifyBase
+from ..URLBase import PrivacyMode
+from ..common import NotifyType
+from ..common import NotifyImageSize
+from ..utils import validate_regex
+from ..utils import parse_bool
+from ..AppriseLocale import gettext_lazy as _
+
+
+class PagerDutySeverity(object):
+    """
+    Defines the Pager Duty Severity Levels
+    """
+    INFO = 'info'
+
+    WARNING = 'warning'
+
+    ERROR = 'error'
+
+    CRITICAL = 'critical'
+
+
+# Map all support Apprise Categories with the Pager Duty ones
+PAGERDUTY_SEVERITY_MAP = {
+    NotifyType.INFO: PagerDutySeverity.INFO,
+    NotifyType.SUCCESS: PagerDutySeverity.INFO,
+    NotifyType.WARNING: PagerDutySeverity.WARNING,
+    NotifyType.FAILURE: PagerDutySeverity.CRITICAL,
+}
+
+
+class NotifyPagerDuty(NotifyBase):
+    """
+    A wrapper for Pager Duty Notifications
+    """
+
+    # The default descriptive name associated with the Notification
+    service_name = 'Pager Duty'
+
+    # The services URL
+    service_url = 'https://pagerduty.com/'
+
+    # Secure Protocol
+    secure_protocol = 'pagerduty'
+
+    # The URL refererenced for remote Notifications
+    notify_url = 'https://events.pagerduty.com/v2/enqueue'
+
+    # A URL that takes you to the setup/help of the specific protocol
+    setup_url = 'https://github.com/caronc/apprise/wiki/Notify_pagerduty'
+
+    # We don't support titles for Pager Duty notifications
+    title_maxlen = 0
+
+    # Allows the user to specify the NotifyImageSize object; this is supported
+    # through the webhook
+    image_size = NotifyImageSize.XY_128
+
+    # Our event action type
+    event_action = 'trigger'
+
+    # Define object templates
+    templates = (
+        '{schema}://{integrationkey}@{apikey}',
+        '{schema}://{integrationkey}@{apikey}/{source}',
+        '{schema}://{integrationkey}@{apikey}/{source}/{component}',
+    )
+
+    # Define our template tokens
+    template_tokens = dict(NotifyBase.template_tokens, **{
+        'apikey': {
+            'name': _('API Key'),
+            'type': 'string',
+            'private': True,
+            'regex': (r'^[a-z0-9_-]+$', 'i'),
```

Comment:
This regex does not work with my API key. It needs to allow for `+` among other characters (perhaps something else as well). I tried to fix this locally but then I get "Could not load Pager Duty URL" when running apprise from command line, not sure why.

Reply:
Interesting, do you know the accepted characters?

I'm wondering... If `@`, `:` or `/` is one of them, i may need to adjust how the url works...

Reply:
I went and generated a bunch of keys and the only symbols I encountered were `-` and `+`. Of course it doesn't prove anything conclusively but I'm pretty confident that's the charset, eg. alphanumeric plus the two symbols.

Reply:
I think i resolved this issue and a few more in the last commit i just pushed.  Would you mind giving it another test for me when you have time?

Reply:
`
apprise -vv -t "Test Message Title" -b "Test Message Body" \
   "pagerduty://***@***/node01.local/drive_sda/"
2022-05-29 23:52:10,684 - INFO - Notifying 1 service(s) asynchronously.
2022-05-29 23:52:11,942 - WARNING - Failed to send Pager Duty notification: Bad Request - Unsupported Parameters., error=400.
`

Still not working for me, but I'm getting another error now.